packages/typespec-ts/src/transform/transformSchemas.ts

```diff
@@ -130,8 +130,9 @@ export function transformSchemas(
         getGeneratedModels(indexer.value, context);
       }
       for (const prop of model.properties) {
+        const [, propType] = prop;
         if (
-          prop[1].type.kind === "Model" &&
+          propType.type.kind === "Model" &&
```

Comment:
why not use propType to replace all the places that has refered to prop[1] ?

Reply:
it should be replaced and will update that